### ISSUE TITLE
Analyzer for CA1822 incorrectly flags use of parameters to a class primary constructor in an instance property getter

### ISSUE THREAD

**Issue by DamianEdwards** [dotnet/roslyn#67783]
**Version Used**:  .NET SDK 8.0.100-preview.3.23178.7

**Steps to Reproduce**:

1. Create a new .NET 8 console project in Visual Studio 2022 17.6
2. Edit the .csproj file to set the language version to preview:
    ```xml
    <LangVersion>preview</LangVersion>
    ```
4. Add the following class definition to *Program.cs*:
    ```csharp
    internal class TestClass(string value)
    {
        public string Value => value;
    }
   ```
5. Note the message from diagnostic CA1822 showing in the Error List

**Diagnostic Id**: CA1822 

If this is a report about a bug in an analyzer, please include the diagnostic if possible (e.g. `"IDE0030"`).

**Expected Behavior**: 

The analyzer does not suggest marking the property as `static`.

**Actual Behavior**:

The analyzer suggests marking the property `static`, which would create a compilation error.
